include/valkey/alloc.h

```diff
@@ -81,11 +82,11 @@ static inline void vk_free(void *ptr) {
 
 #else
 
-void *vk_malloc(size_t size);
-void *vk_calloc(size_t nmemb, size_t size);
-void *vk_realloc(void *ptr, size_t size);
-char *vk_strdup(const char *str);
-void vk_free(void *ptr);
+LIBVALKEY_API void *vk_malloc(size_t size);
```

Comment:
Please remind me, why are the malloc functions in the public API?

Reply:
Becaused you can [replace them with your own allocators](https://github.com/valkey-io/libvalkey/blob/f670346b9d28f5c957c7457112faca4731edd503/include/valkey/alloc.h#L50)

This is really useful if you want to use the library under the hood in some larger system that uses a custom allocator.

Reply:
Hiding them is fine by me. Looking through the header again I think I did it that way because then you can make the wrappers that call the custom allocators inline. 

I can't imagine the performance difference will be meaningful if we hide them.

Reply:
Good point actually. We can keep them for inlining. :)